### PR TITLE
DB-1721: Code cleanup in db-engine

### PR DESCRIPTION
**Pull request by Ben-Epstein**

### CODE REVIEW

File: db-engine/src/main/java/com/splicemachine/db/catalog/types/IndexDescriptorImpl.java [hunk 1]
```diff
@@ -168,22 +168,16 @@ public String indexType()
...
-	public boolean			isAscending(Integer keyColumnPosition)
-	{
-		int i = keyColumnPosition.intValue() - 1;
-		if (i < 0 || i >= baseColumnPositions.length)
-			return false;
-		return isAscending[i];
-	}
+	public boolean			isAscending(Integer keyColumnPosition) {
+        int i = keyColumnPosition - 1;
+        return !(i < 0 || i >= baseColumnPositions.length) && isAscending[i];
```
Comment (dgomezferro) [inline db-engine/src/main/java/com/splicemachine/db/catalog/types/IndexDescriptorImpl.java]: This is harder to understand IMO.

File: db-engine/src/main/java/com/splicemachine/db/catalog/types/IndexDescriptorImpl.java [hunk 2]
```diff
@@ -168,22 +168,16 @@ public String indexType()
...
-	public boolean			isDescending(Integer keyColumnPosition)
-	{
-		int i = keyColumnPosition.intValue() - 1;
-		if (i < 0 || i >= baseColumnPositions.length)
-			return false;
-		return ! isAscending[i];
-	}
+	public boolean			isDescending(Integer keyColumnPosition) {
+        int i = keyColumnPosition - 1;
+        return !(i < 0 || i >= baseColumnPositions.length) && !isAscending[i];
```
Comment (dgomezferro) [inline db-engine/src/main/java/com/splicemachine/db/catalog/types/IndexDescriptorImpl.java]: Same.

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/BinaryLogicalOperatorNode.java [hunk 3]
```diff
@@ -100,18 +100,9 @@ public ValueNode bindExpression(FromList fromList,
...
-			return (leftOperand.verifyEliminateNots() &&
-					rightOperand.verifyEliminateNots());
-		}
-		else
-		{
-			return true;
-		}
-	}
+	boolean verifyEliminateNots() {
+        return !SanityManager.ASSERT || (leftOperand.verifyEliminateNots() && rightOperand.verifyEliminateNots());
```
Comment (dgomezferro) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/BinaryLogicalOperatorNode.java]: I would keep this pattern as it was, it's easier to see what's part of the assertion and what's not. If there are other cases please revert them too.

**Comment by snowspeeder** [comment]
Ben, could you please incorporate Daniel's comment and refresh PR

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/BinaryLogicalOperatorNode.java (hunk 3 above)
Comment (Ben-Epstein) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/BinaryLogicalOperatorNode.java]: Reverted

File: db-engine/src/main/java/com/splicemachine/db/catalog/types/IndexDescriptorImpl.java (hunk 2 above)
Comment (Ben-Epstein) [inline db-engine/src/main/java/com/splicemachine/db/catalog/types/IndexDescriptorImpl.java]: Reverted

File: db-engine/src/main/java/com/splicemachine/db/catalog/types/IndexDescriptorImpl.java (hunk 1 above)
Comment (Ben-Epstein) [inline db-engine/src/main/java/com/splicemachine/db/catalog/types/IndexDescriptorImpl.java]: Reverted

**Pull request closed without merging**